src/app/(dashboard)/dashboard/chat/[chatId]/page.tsx

```diff
@@ -10,12 +37,65 @@ interface ChatProps{
 
 const Page: FC<ChatProps> = async ({params}) => {
     const session = await myGetServerSession();
-    if (!session || !params.chatId.split('--').includes(session.user.id)){
+    const userId = session?.user?.id as string
+    const {chatId} = params
+    const participants = new Participants(chatId, userId as string)
+    const helpers = new Helpers()
+    const initialMessages = await helpers.getChatMessages(chatId)
+
+    if (!(session && participants.includesSession())){
```

Comment:
don't need to evaluate session here. If session is null, then User ID will be an empty string.Add a test case for this if it doesn't exist already and see if that check can be removed